Sources/TuistGenerator/GraphMappers/DeleteDerivedDirectoryGraphMapper.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+import TuistCore
+import TuistSupport
+
+/// A mapper that returns side effects to delete the derived directory of each project of the graph.
+public class DeleteDerivedDirectoryGraphMapper: GraphMapping {
+    public init() {}
+
+    // MARK: - GraphMapping
+
+    public func map(graph: Graph) throws -> (Graph, Set<SideEffectDescriptor>) {
+        logger.debug("Determining the /Derived directories that should be delted")
+        var sideEffects = Set<SideEffectDescriptor>()
+
+        graph.projects.forEach { project in
```

Comment:
this whole mappers concept is turning out to be very beautiful!

Reply:
It is. I love it a lot ❤️